src/pudl/etl/epacems_assets.py

```diff
@@ -163,15 +123,17 @@ def core_epacems__hourly_emissions(
     information see:
     https://docs.dagster.io/concepts/ops-jobs-graphs/dynamic-graphs.
     """
-    years = get_years_from_settings()
-    partitions = years.map(
-        lambda year: process_single_year(
-            year,
-            _core_epa__assn_eia_epacamd_unique,
-            core_eia__entity_plants,
+    year_quarters = get_year_quarters_from_settings()
+    partitions = year_quarters.map(
+        lambda year_quarter: extract_quarter(
+            year_quarter,
         )
     )
-    return consolidate_partitions(partitions.collect())
+    return transform_and_write_monolithic(
+        partitions.collect(),
```

Comment:
Is this the step that actually triggers the evaluation?

Reply:
This is actually collecting all of the `op`s for dagster. They are split into two steps right now that get triggered by the `sink_parquet` method. Right now the way it works is creating a bunch of ops to extract one quarter at a time, because you have to read the CSVs into memory to get them out of the zipfiles they're archived in, so there's no memory efficient way to do the whole extract as together. The transform is then done in one go though, as Polars can stream from the parquet files on disk.